airflow/bin/cli.py

```diff
@@ -339,6 +339,16 @@ def run(args, dag=None):
         settings.configure_vars()
         settings.configure_orm()
 
+    # Get task attempt that will be used to separate log based on
+    # each attempt number
+    TI = TaskInstance
+    session = settings.Session()
+    ti = session.query(TI).filter(
+        TI.dag_id == args.dag_id,
+        TI.task_id == args.task_id,
+        TI.execution_date == args.execution_date).first()
+    attempt = ti.attempt + 1 if ti else 1
```

Comment:
What are you doing with attempt? It is not used. The session is not committed and not closed. 

In addition if you want to update the field make sure to read up on sqlalchemy's update mechanism in order not to end up with race conditions.